### ISSUE TITLE
Join data "on" only automatically populated for upward relationships

### ISSUE THREAD

**Issue by willbryant** [metabase/metabase#23819]
**Describe the bug**
If you have a 1..n relationship such as cities to purchases, when you start a question with purchases and then use "Join data" to add the cities table to the query, the "on" join criteria are automatically populated with the join relationship set up as an FK in the data model.

However, if you happen to do it the other way around - start a question with cities and then use "Join data" to add the purchases table to the query, the "on" join criteria is not populated.

This inconsistency is quite confusing to users, and they then need to learn how join criteria work, or be taught to start again with the leaf query (throwing away work they've already done like selecting the columns, in the process).

**To Reproduce**
1. Start a question on the "Sample database"
2. Choose the "Products" table
3. Add a join to the "Orders" table
4. Note that the join criteria has not been populated.
5. Repeat in the other order. Note that the join criteria *has* been populated this way around.

**Expected behavior**
When starting a question with a "parent" table and joining a 1..n "child" table, the join criteria should automatically be populated from the available FK relationships, the same way as happens when you join in the other order.

**Screenshots**

**Information about your Metabase Installation:**

You can get this information by going to Admin -> Troubleshooting.

- Your browser and the version: Chrome 103
- Your operating system: macOS 12.3.1
- Your databases: Redshift
- Metabase version: 0.43.3
- Metabase hosting environment: Docker
- Metabase internal database: H2

**Severity**
How severe an issue is this bug to you? Is this annoying, blocking some users, blocking an upgrade or blocking your usage of Metabase entirely?

Moderately severe, one of a set of UX issues that are pushing us towards competing products to support our non-technical users better.

**Comment by flamber** [comment]
Duplicate of #1177

**Comment by willbryant** [comment]
@flamber I don't think this is same thing? I read that as being more about the right sidebar when you're looking at a table, automatically exposing the fields from the joined table so they're available to filter/summarise.

This is just about when you use the join option explicitly on the new question page and add a table to the question, filling in the join fields automatically.

**Comment by flamber** [comment]
It would still require exposing connections on both sides, which we currently don't support, so this is not a bug.
